Chp12Lsn7/UserCommunication/UserBarrier.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace UserCommunication
+{
+    public class UserBarrier
+    {
+        static void Main(string[] args)
+        {
+        }
+
+        /// <summary>
+        /// Takes a positive integer from the user and returns it.
+        /// </summary>
+        /// <param name="userInput">A null string. Value is changed within method. Allows for unit testing.</param>
+        /// <returns>A positive integer</returns>
+        public static int UserInput(string userInput)
+        {
+            Console.WriteLine("Please enter a positive integer greater than zero.");
+            Console.WriteLine("");
+            Console.Write("> ");
+
+            if (userInput == null)
+            {
+                userInput = Console.ReadLine();
+            }
+            int userNumber = 0;
+
+            try
+            {
+                userNumber = Convert.ToInt32(userInput);
+
+                if (userNumber <= 0)
+                {
+                    throw new ArgumentException("Invalid Input! Cannot enter a number less than or equal to zero.");
+                }
+
+            }
+            finally
```

Comment:
This `finally` normally would be incorrect.  It will run both when there is a *success* and when an exception is thrown.  Do you really want to say "thank you" when an exception is thrown?